### PR TITLE
Fix CATA Feedback + Activity Hints

### PR DESCRIPTION
**Pull request by LEDThereBeLight**
Pulled out the bug fixes from a larger activity testing branch since I ran into some test setup problems that I need to fix.

Closts #912 

**Bug Fixes**
- Check all that apply: fix feedback/score not showing up in delivery
- Hints: fix hint requests and display in all activities when empty hints are present

### CODE REVIEW

**Comment by LEDThereBeLight** [comment]
There's a merge from master issue here causing a build failure - fixing it

**Comment by eliknebel** [comment]
Screen Shot 2021-04-14 at 10 57 19 AM

I'm still seeing an issue with no feedback in the activity live preview. Was this supposed to fix that as well?

File: assets/src/components/resource/TestModeHandler.tsx [hunk 1]
```diff
@@ -99,28 +100,28 @@ export class TestModeHandler extends React.Component<TestModelHandlerProps, Test
   // we can handle individual part and activity submits in a unified manner
   handleSubmit(continuation: Continuation, partId: string, partInputs: PartResponse[]) {
     Persistence.evaluate(this.state.model, partInputs)
-    .then((result: Persistence.Evaluated) => {
```
Comment (eliknebel) [inline assets/src/components/resource/TestModeHandler.tsx:102]: This is the fix for the feedback in live preview that should remain intact

**Review by eliknebel — CHANGES_REQUESTED** [review]

File: assets/src/components/resource/TestModeHandler.tsx [hunk 2]
```diff
@@ -129,14 +130,15 @@ export class TestModeHandler extends React.Component<TestModelHandlerProps, Test
         };
       });
 
-    continuation({ type: 'success', actions: evaluatedParts }, undefined);
```
Comment (eliknebel) [inline assets/src/components/resource/TestModeHandler.tsx:132]: this must remain named 'actions' as well

File: assets/src/components/resource/TestModeHandler.tsx [hunk 3]
```diff
@@ -99,28 +100,28 @@ export class TestModeHandler extends React.Component<TestModelHandlerProps, Test
...
-            type: 'FeedbackAction',
-            error: e.error,
-            attempt_guid: e.part_id,
-            out_of: e.result.out_of,
-            score: e.result.score,
-            feedback: e.feedback,
-          };
-        });
-
-      continuation({ type: 'success', actions }, undefined);
```
Comment (eliknebel) [inline assets/src/components/resource/TestModeHandler.tsx:116]: This should remain named 'actions'

**Review by eliknebel — CHANGES_REQUESTED** [review]

File: assets/src/components/resource/TestModeHandler.tsx [hunk 4]
```diff
@@ -99,28 +100,28 @@ export class TestModeHandler extends React.Component<TestModelHandlerProps, Test
   // we can handle individual part and activity submits in a unified manner
   handleSubmit(continuation: Continuation, partId: string, partInputs: PartResponse[]) {
     Persistence.evaluate(this.state.model, partInputs)
-    .then((result: Persistence.Evaluated) => {
-
-      const actions = result.evaluations
-        .map((e: any) => {
-          return {
-            type: 'FeedbackAction',
```
Comment (darrensiegel) [inline assets/src/components/resource/TestModeHandler.tsx:107]: This should remain "FeedbacAction"

**Pull request closed without merging**